Drivers/USBClient/USBClient.cpp

```diff
@@ -0,0 +1,1328 @@
+// Copyright Microsoft Corporation
+// Copyright Oberon microsystems, Inc
+// Copyright GHI Electronics, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+#include "USBClient.h"
+
+#define __min(a,b)  (((a) < (b)) ? (a) : (b))
+
+TinyCLR_UsbClient_DataReceivedHandler UsbClient_DataReceivedHandler;
+TinyCLR_UsbClient_OsExtendedPropertyHandler UsbClient_OsExtendedPropertyHandler;
+
+// usb fifo buffer
+static int32_t usb_packet_fifo_in[CONCAT(DEVICE_TARGET, _USB_ENDPOINT_COUNT)];
```

Comment:
Don't use the CONCAT(DEVICE_TARGET, ...) pattern in this file. Instead add a function like `STM32F4_UsbClient_GetConfiguration(name, manufacturer, fifoCount, endpointSize, ...)`